### DISCUSSION TITLE
How to identify column groups?

### DISCUSSION THREAD

**Discussion by cool-RR** [saulpw/visidata#1694] [Q&A] [answered]
I'm experimenting with column groups. One thing I want to do is a variant of the `g_` command that doesn't expand the width of column groups to an insane width. How can I identify whether one of the columns in `sheet.visibleCols` is a column group?

**Comment by saulpw** [comment, marked as answer]
You can't tell if a column itself is a column group; it's just if the cells in that column contain a dict (or a list).  You can tell if a column is an expanded column (part of a group), if it has the `origCol` property (though some other derived but non-expanded columns may have that property too).

If you find a better algorithm for `g_`, please post it!

**Reply by cool-RR** [reply]
That's what I figured. My algorithm samples the first row to find if the column is a group; I'm guessing this is considered a workaround.

Anyway, this is my code:

```
    for column in sheet.visibleCols:
        column: visidata.Column
        if column.type is anytype:
            try:
                first_value = column.getValue(sheet.rows[0])
            except Exception:
                # I got an inexplicable `NameError` here while doing this on a `DirSheet`, so let's
                # not guess a column when that happens.
                continue
            if isinstance(first_value, dict):
                column.setWidth(len(column.name) + 5)
                continue
        column.setWidth(column.getMaxWidth(sheet.visibleRows))
```
